src/databricks/labs/ucx/workspace_access/manager.py

```diff
@@ -144,6 +149,35 @@ def apply_group_permissions(self, migration_state: MigrationState) -> bool:
         logger.info("Permissions were applied")
         return True
 
+    def apply_group_permissions_private_preview_api(self, migration_state: MigrationState) -> bool:
+        if len(migration_state) == 0:
+            logger.info("No valid groups selected, nothing to do.")
+            return True
+        logger.info(
+            f"Applying the permissions to account groups. "
+            f"Total groups to apply permissions: {len(migration_state)}."
+        )
+
+        def migrate_permission(workspace_id, workspace_group_name, account_group_name):
+            self._ws.permission_migration.migrate_permissions(workspace_id, workspace_group_name, account_group_name)
+
+        applier_tasks: list[Callable[..., None]] = []
```

Comment:
this should be sequential, based on the design doc and we just need to check the amount of migrated/remaining records.

Reply:
done